pallets/marketplace/src/types.rs

```diff
@@ -125,6 +125,21 @@ pub struct Assignment<Reward> {
     pub acknowledged: bool,
     /// Keeps track of the SLA.
     pub sla: SLA,
+    /// Processor Pub Keys
+    pub pub_keys: Option<PubKeys>,
```

Comment:
Option should not be necessary here since we can just add an empty vector.
```suggestion
    pub pub_keys: PubKeys,
```